File: AppointmentSchedulerAPI/layers/CrossCuttingLayer/Communication/Model/ApiResponse.cs
namespace AppointmentSchedulerAPI.layers.CrossCuttingLayer.Communication.Model
{
    public class ApiResponse<T>
    {
        public int Status { get; set; }
        public string Message { get; set; }
        public T? Data { get; set; }
        public string Version { get; set; }

        public ApiResponse(int status, string message, string version, T data)
        {
            Status = status;
            Message = message;
            Data = data;
            Version = version;
        }
        public ApiResponse(int status, string message, string version)
        {
            Status = status;
            Message = message;
            Version = version;
        }

    }
}

File: AppointmentSchedulerWeb/src/cross-cutting/communication/model/api-response.ts
import { ApiVersionType } from "./api-version.types";
import { MessageCodeType } from "./message-code.types";

export interface ApiSuccessResponse<TData> {
  status: number;
  message: MessageCodeType;
  data: TData,
  version: ApiVersionType;
}

export interface ApiErrorResponse<TError> {
  status: number;
  message: MessageCodeType;
  data: TError,
  version: ApiVersionType;
}

export type ApiResponse<TData, TError> = ApiSuccessResponse<TData> | ApiErrorResponse<TError>;
